model2vec/inference/model.py

```diff
@@ -0,0 +1,141 @@
+from __future__ import annotations
+
+import re
+from pathlib import Path
+from tempfile import TemporaryDirectory
+
+import huggingface_hub
+import numpy as np
+import skops.io
+from sklearn.neural_network import MLPClassifier
+from sklearn.pipeline import Pipeline
+
+from model2vec.model import PathLike, StaticModel
+
+_DEFAULT_TRUST_PATTERN = re.compile(r"sklearn\..+")
+_DEFAULT_MODEL_FILENAME = "pipeline.skops"
+
+
+class StaticModelPipeline:
+    def __init__(self, model: StaticModel, head: Pipeline) -> None:
+        """Create a pipeline with a StaticModel encoder."""
+        self.model = model
+        self.head = head
+
+    @classmethod
+    def from_pretrained(
+        cls: type[StaticModelPipeline], path: PathLike, token: str | None = None
+    ) -> StaticModelPipeline:
+        """
+        Load the pipeline from the trained model.
+
+        :param path: The path to the folder containing the pipeline.
+        :param token: The token to use to download the pipeline from the hub.
+        :return: The loaded pipeline.
+        """
+        model, head = _load_pipeline(path, token)
+        model.embedding = np.nan_to_num(model.embedding)
+
+        return cls(model, head)
+
+    def save_pretrained(self, path: str) -> None:
+        """Save the model to a folder."""
+        save_pipeline(self, path)
+
+    def push_to_hub(self, repo_id: str, token: str | None = None, private: bool = False) -> None:
```

Comment:
I would add a modelcard and perhaps tags or a library reference, this helps a lot with visibility, usability and findability.

https://huggingface.co/docs/hub/model-cards#specifying-a-library

Reply:
This actually already happens because we push the underlying static model to the hub, which has a model card. This model card template is specified in the root of the code.